src/rust/zh.dmzj/src/lib.rs

```diff
@@ -0,0 +1,439 @@
+				.unwrap()
+				.strip_suffix(';')
+				.unwrap();
+
+			json::parse(r.as_bytes()).as_array()?
+		};
+
+		for it in data {
+			let it = it.as_object()?;
+			let id = helper::i32_to_string(it.get("id").as_int()? as i32);
+			let author = it.get("comic_author").as_string()?.read();
+			let title = it.get("comic_name").as_string()?.read();
+			let cover = it.get("comic_cover").as_string()?.read();
+
+			manga_arr.push(Manga {
+				id,
+				cover,
+				title,
+				author: String::new(),
+				artist: author,
+				description: String::new(),
+				url: String::new(),
+				categories: Vec::new(),
+				status: MangaStatus::Unknown,
+				nsfw: MangaContentRating::Safe,
+				viewer: MangaViewer::Rtl,
+			});
+		}
+	} else {
+		let mut filters_query: String = String::from("0");
+		if !filters_list.is_empty() {
+			for i in filters_list {
+				filters_query.push_str(&helper::i32_to_string(i));
+				filters_query.push('-');
+			}
+			filters_query.pop();
+		}
+
+		let url = format!(
+			"{}/classify/{}/{}/{}.json",
+			V3_API_URL,
+			filters_query,
+			helper::i32_to_string(sort),
+			helper::i32_to_string(page)
+		);
+
+		let data = Request::new(&url, HttpMethod::Get).json().as_array()?;
+
+		for it in data {
+			let it = it.as_object()?;
+			let id = helper::i32_to_string(it.get("id").as_int()? as i32);
+			let author = it.get("authors").as_string()?.read();
+			let title = it.get("title").as_string()?.read();
+			let cover = it.get("cover").as_string()?.read();
+			let status = match it.get("status").as_string()?.read().as_str() {
+				"连载中" => MangaStatus::Ongoing,
+				"已完结" => MangaStatus::Completed,
+				_ => MangaStatus::Unknown,
+			};
+			manga_arr.push(Manga {
+				id,
+				cover,
+				title,
+				author: String::new(),
+				artist: author,
+				description: String::new(),
+				url: String::new(),
+				categories: Vec::new(),
+				status,
+				nsfw: MangaContentRating::Safe,
+				viewer: MangaViewer::Rtl,
+			});
+		}
+	}
+
+	let len = manga_arr.len();
+	Ok(MangaPageResult {
+		manga: manga_arr,
+		has_more: if is_keyword { false } else { len != 0 },
+	})
+}
+
+#[get_manga_details]
+fn get_manga_details(id: String) -> Result<Manga> {
+	let url = format!(
+		"{}/comic/detail/{}?channel=android&version=3.0.0&timestamp={}",
+		V4_API_URL,
+		&id,
+		aidoku::std::current_date() as i64
+	);
+
+	let pb = helper::decode(&helper::get(url).string());
+	if pb.errno == 0 {
+		let pb_data = pb.data.unwrap();
+		return Ok(Manga {
+			id: id.clone(),
+			cover: pb_data.cover,
+			title: pb_data.title,
+			author: String::new(),
+			artist: String::new(),
+			description: pb_data.description,
+			url: format!("{}/info/{}.html", BASE_URL, id),
+			categories: pb_data.types.iter().map(|s| s.tag_name.clone()).collect(),
+			status: match pb_data.status[0].tag_name.as_str() {
+				"连载中" => MangaStatus::Ongoing,
+				"已完结" => MangaStatus::Completed,
+				_ => MangaStatus::Unknown,
+			},
+			nsfw: MangaContentRating::Safe,
+			viewer: MangaViewer::Ltr,
```

Comment:
```suggestion
			viewer: MangaViewer::Rtl,
```

Reply:
Chinese traditional reading direction is left to right, and don't have enough info to identify whether image is long enough that should read it is vertical. So just assume default direction is ltr.

Reply:
Yeah, english and most other languages' reading direction is left to right, but manga is still read right to left because, even though inside the text bubbles is left to right, the order you travel between panels is right to left.

I see that you can filter which location a title comes from (like japan is 日本 and would be rtl, south korea is 韩国 and would be vertical scroll, etc.), so that's how you could assume reading direction.

Reply:
You are right. Fixed.